tasks/build/os_shell_scripts.js

```diff
@@ -0,0 +1,43 @@
+import {unlink} from 'fs';
+import {join, extname} from 'path';
+import {promisify} from 'bluebird';
+import {ncp} from 'ncp';
+const pncp = promisify(ncp);
+const punlink = promisify(unlink);
+
+export default function (grunt) {
+  grunt.registerTask('_build:osShellScripts', function () {
+    const done = this.async();
+    const source = 'build/kibana/bin';
+    const platforms = grunt.config.get('platforms');
+    const allPlatforms = fn => invokeAllAsync(platforms, fn);
+
+    run()
+      .then(() => done())
+      .catch(e => done(e));
+
+    async function run() {
+      await allPlatforms(platform => punlink(join(platform.buildDir, 'bin')));
+      await allPlatforms(platform => pncp(source, join(platform.buildDir, 'bin')));
+      await allPlatforms(removeExtraneousShellScripts);
+    }
+  });
+
+  function removeExtraneousShellScripts(platform) {
+    return Promise.all(grunt.file
+      .expand(join(platform.buildDir, 'bin', '*'))
+      .filter(isExtraneous)
+      .map(file => punlink(file)));
+
+    function isExtraneous(file) {
```

Comment:
I have the same thoughts on this as I did with the `allPlatforms` closure in the last round of feedback: can we create it as a top level function that takes in `platform` as an argument rather than putting all of the logic in a closure?  If you still want to pass a function to .filter() without using an arrow function, you can always partially apply it before beginning your promise chain.

So basically:

```js
function isExtraneous(platform, file) {
  const ext = extname(file)
  if (platform.win && ext === '') { return true; }
  if (!platform.win && ext === '.bat') { return true; }
  return false;
}

function removeExtraneousShellScripts(platform) {
  // return Promise.all etc.
    .filter(file => isExtraneous(platform, file))
}
// or
function removeExtraneousShellScripts(platform) {
  const isExtraneousOnPlatform = file => isExtraneous(platform, file);
  // return Promise.all etc.
    .filter(isExtraneousOnPlatform)
}
```

As I said in the previous feedback, this is a personal preference, but you seemed to agree with it before.